### PR TITLE
hourly comparison interface

### PR DESCRIPTION
**Pull request by TrystanLea**

### CODE REVIEW

File: www/views/compare.html [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+<body>
+
+<div id="header">
+  <div id="title"><span class="big">
+    <b>HeatpumpMonitor</b>.org</span></div>
+  <div id="navigation">
+    <a href="." title="Home"><i aria-hidden="true" class="fas fa-home"></i></a>
+    <a href="stats.html" title="30 Day Stats"><i aria-hidden="true" class="fas fa-table"></i></a>
+    <a href="costs.html" title="Running Costs"><i aria-hidden="true" class="fas fa-coins"></i></a>
+    <a href="graph.html" title="Comparison Charts" class="active"><i aria-hidden="true" class="fas fa-chart-line"></i></a>
```
Comment (MrTimbones) [inline www/views/compare.html:22]: We're going to have to split this into two links, and think of better titles, or we have all the chart features on one page.

The hardest part is picking an appropriate icon. I'm picking from [fontawesome v5.15](https://fontawesome.com/v5/search?o=r&s=solid), and best I can find is `<i class="fas fa-object-group"></i>`. v6 has fa-chart-scatter, but I couldn't get it to work.

File: www/views/compare.html [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+  
+  <div id="gd" style="height:650px"></div>
+  
+
+<div class="footer">An <a href="https://openenergymonitor.org/">OpenEnergyMonitor.org</a> community initiative</div>
+</body>
+</html>
+
+
+<script>
```
Comment (MrTimbones) [inline www/views/compare.html:60]: Should we move this script into its own file?

**Review by MrTimbones — CHANGES_REQUESTED** [review]

**Comment by MrTimbones** [comment]
Can we make the 2nd system selector default to `None` so that people can view a single system if they wish, with the 2nd system being optional.

**Future idea**: allow selection of the same system on both sides, but with different dates.

File: scrape_kwh.php [hunk 3]
```diff
@@ -20,6 +20,7 @@
   # read each row and append kWh readings for last year
   while (($row = fgetcsv($input, 1000, "\t")) !== FALSE) {
     $system = ['id' => count($data) + 1];
+    echo json_encode($system)."\n";
```
[the remark this answers is not in the document]
Comment (TrystanLea) [reply scrape_kwh.php:23]: they are not essential, Id like to cut down the number of requests made here, I think we could get more from the stats api.. will look into it.

File: www/views/compare.html (hunk 1 above)
Comment (TrystanLea) [reply www/views/compare.html:24]: ok sounds good, I think getting all the chart features on the same page may be too much of a challenge for now. 
I think there will be another graph page shortly to provide the half hourly profile for all systems...

File: www/views/compare.html (hunk 2 above)
Comment (TrystanLea) [reply www/views/compare.html:83]: sure will do that

File: scrape_kwh.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (MrTimbones) [reply scrape_kwh.php:23]: Are you worried about load on the server as the number of systems grows? This script is only really run once-a-day, around midnight, and we can put in some sleeps to spread the load if that will help.

**Comment by TrystanLea** [comment]
Hello @MrTimbones 

I've moved the header, navigation and footer for each page to a theme file and moved the javascript out into js files.
Let me know if you are happy for me to merge this.

Cheers!

**Comment by MrTimbones** [comment]
I can't get this branch to run on my test server, but I guess it works fine for you?

**Comment by TrystanLea** [comment]
yes works fine here, just requires modrewrite enabled and php (same requirements as for emoncms). Do you see any errors? any clues as to why it's not working?

**Comment by MrTimbones** [comment]
I'll figure it out. If it works for you and you're confident that it'll work on the main server, I'm happy for this to be merged,

**Comment by TrystanLea** [comment]
ok great, thanks!

**Comment by TrystanLea** [comment]
Ok turns out I had missed pushing up index.php! as it was in .gitignore :) fixed now

**Pull request merged**